crates/subspace-core-primitives/src/lib.rs

```diff
@@ -1075,8 +1172,9 @@ impl SectorId {
         &self,
         global_challenge: &Blake3Hash,
     ) -> SectorSlotChallenge {
-        let sector_slot_challenge = Simd::from(self.0) ^ Simd::from(*global_challenge);
-        SectorSlotChallenge(sector_slot_challenge.to_array())
+        let sector_slot_challenge =
+            Simd::from(self.0.as_array()) ^ Simd::from(global_challenge.as_array());
```

Comment:
Too verbose:
```suggestion
        let sector_slot_challenge = Simd::from(*self.0) ^ Simd::from(*global_challenge);
```

Reply:
It's actually:
```suggestion
        let sector_slot_challenge =
            Simd::from(*self.0) ^ Simd::from(**global_challenge);
```

How do you feel about double derefs?

Reply:
I think it is fine. We could create a custom SIMD impl for `GlobalChallenge`, but I don't think it is intuitive, so double deref is good enough.